src/sardana/macroserver/scan/gscan.py

```diff
@@ -2277,7 +2277,13 @@ def _go_through_waypoints(self):
             self.data.initial_data = initial_data
 
             if hasattr(macro, 'getHooks'):
-                for hook in macro.getHooks('pre-start'):
+                pre_acq_hooks = macro.getHooks('pre-start')
+                if len(pre_acq_hooks) > 0:
+                    self.macro.warning("pre-start hook place is deprecated,"
```

Comment:
Here you are using the macro logging stream. This will appear in spock. I don't mind. For sure this will be more efficient to get rid of the deprecated calls. Just a comment ...

Reply:
We have moved the hooks execution after the try-except-finally block, so they will be executed only if no exception occurred.